src/util/util_watchdog.c

```diff
@@ -264,3 +271,20 @@ int get_watchdog_ticks(void)
 {
     return __sync_add_and_fetch(&watchdog_ctx.ticks, 0);
 }
+
+void arm_watchdog(void)
+{
+    if (watchdog_ctx.armed) {
+        DEBUG(SSSDBG_CRIT_FAILURE,
```

Comment:
I was going to ask to make `arm_watchdog()` return bool (success/fail) but after some thought I agree catching this logic error right in the function is better.

Reply:
Hi,

I was thinking about this as well, but then the question came, what would the caller do if the call returns fail?

bye,
Sumit